### DISCUSSION TITLE
How can I update Wazuh components and agents

### DISCUSSION THREAD

**Discussion by styxiner** [wazuh/wazuh-docker#1289] [Q&A] [answered]
So, I'm on v4.3.10, and I want to update to the current version v4.7.3. The problem is that I'm on docker and I don't know how I should do it. Anyone help? If Wazuh's main components are updated, it sends an auto-update to the agents??
![Captura de pantalla 2024-04-08 162258](https://github.com/wazuh/wazuh-docker/assets/160549592/993ec7ee-3af6-49b4-ac36-1ac37934b946)

**Comment by rauldpm** [comment, marked as answer]
Hello

The steps may vary slightly depending on your current deployment.

- Is it a single-node or a multi-node deployment
- Have you modified docker-compose with custom modifications?

Assuming it is a single node and you have no modifications in the docker-compose, you can follow the following documentation: https://documentation.wazuh.com/current/deployment-options/docker/upgrading-wazuh-docker.html#upgrading-wazuh-docker

Keep in mind that this guide offers two options that should not be executed sequentially since they are for different cases, in case you do not have modifications, refer to the previous guide (basically only the following commands)

- docker-compose down
- git checkout v4.7.3
- docker-compose up -d

If you have made custom modifications in the docker-compose file and want to maintain it, you must perform the steps specified in https://documentation.wazuh.com/current/deployment-options/docker/upgrading-wazuh-docker.html#keeping-custom-docker-compose-files (the section following the previous one), keep in mind that this section does not execute the git checkout command, the repository version is maintained and only the docker-compose file is updated with the indicated changes

When testing, I found that part of the configuration is missing, which I have reported here https://github.com/wazuh/wazuh-documentation/issues/7231, if this is your case please compare the configuration and add the changes that I indicate in the issue images, for convenience, I share with you the docker-compose of v4.7.3

Also, the agents are not upgraded automatically, you have to upgrade them manually or using API calls to upgrade them to the Wazuh manager version, I share with you the manual and remote upgrade documentation

- https://documentation.wazuh.com/current/upgrade-guide/index.html
- https://documentation.wazuh.com/current/user-manual/agent/agent-management/remote-upgrading/upgrading-agent.html

**Reply by rauldpm** [reply]
Closed due to inactivity
